python/lsst/summit/utils/nightReport.py

```diff
@@ -0,0 +1,633 @@
+            pass ``observation_type='science'``.
+        """
+        seqNums = self.data.keys() if not kwargs else self.getSeqNumsMatching(**kwargs)
+        seqNums = sorted(seqNums)  # should always be sorted, but is a totaly disaster here if not
+
+        dts = self.getTimeDeltas()
+        lines = []
+        for seqNum in seqNums:
+            try:
+                expTime = self.data[seqNum]['exposure_time'].value
+                imageType = self.data[seqNum]['observation_type']
+                target = self.data[seqNum]['target_name']
+                deadtime = dts[seqNum]
+                filt = self.data[seqNum]['physical_filter']
+
+                msg = f'{seqNum} {target} {expTime:.1f} {deadtime:.02f} {imageType} {filt}'
+            except Exception:
+                msg = f"Error parsing {seqNum}!"
+            lines.append(msg)
+
+        print(r"seqNum target expTime deadtime imageType filt")
+        print(r"------ ------ ------- -------- --------- ----")
+        for line in lines:
+            print(line)
+
+    def getExposureMidpoint(self, seqNum):
+        """Return the midpoint of the exposure as a float in MJD.
+
+        Parameters
+        ----------
+        seqNum : `int`
+            The seqNum to get the midpoint for.
+
+        Returns
+        -------
+        midpointMjd : `float`
+            The midpoint, as an mjd float.
+        """
+        timespan = self.data[seqNum]['timespan']
+        return (timespan.begin.mjd + timespan.end.mjd) / 2
+
+    def plotPerObjectAirMass(self, objects=None, airmassOneAtTop=True, saveFig=''):
+        """Plot the airmass for objects observed over the course of the night.
+
+        Parameters
+        ----------
+        objects : `list` [`str`], optional
+            The objects to plot. If not provided, all objects are plotted.
+        airmassOneAtTop : `bool`, optional
+            Put the airmass of 1 at the top of the plot, like astronomers
+            expect.
+        saveFig : `str`, optional
+            Save the figure to this file path?
+        """
+        if not objects:
+            objects = self.stars
+
+        objects = ensure_iterable(objects)
+
+        _ = plt.figure(figsize=(10, 6))
+        for star in objects:
+            seqNums = self.getSeqNumsMatching(target_name_short=star)
+            airMasses = [self.data[seqNum]['boresight_airmass'] for seqNum in seqNums]
+            obsTimes = [self.getExposureMidpoint(seqNum) for seqNum in seqNums]
+            color = self.cMap[star].color
+            marker = self.cMap[star].marker
+            plt.plot(obsTimes, airMasses, color=color, marker=marker, label=star, ms=10, ls='')
+
+        plt.ylabel('Airmass', fontsize=20)
+        plt.ylabel('MJD', fontsize=20)
+        if airmassOneAtTop:
+            ax = plt.gca()
+            ax.set_ylim(ax.get_ylim()[::-1])
+        _ = plt.legend(bbox_to_anchor=(1, 1.025), prop={'size': 15}, loc='upper left')
```

Comment:
You don't have to make plt.legend = anything if you don't care about it.

Reply:
Thanks, I think doing `_ = ` for matplotlib calls is a hangover from notebook development where if you don't catch it, and you do nothing else in the cell, it prints out what it returned, which if it's a histogram with a thousand patches, is super lame. Bad habits 🙃